tests/test_api_auth.py

```diff
@@ -0,0 +1,92 @@
+"""Testing all things authentication through the API."""
+from base64 import b64encode
+import json
+import unittest
+
+from flask import url_for
+
+from app import create_app, db
+from app.api import auth
+from app.models import User
+
+
+class ApiAuthentication(unittest.TestCase):
+    """Test all authentication through the API."""
+
+    def setUp(self):
+        """Setup app for testing before each test case."""
+        self.app = create_app('testing')
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+        db.create_all()
+        self.client = self.app.test_client(use_cookies=True)
+        self.user = User(first_name='ichiato', last_name='ikikin',
+                         email='ichiato@yahoo.com', password='password')
+
+    def tearDown(self):
+        """Delete app and db instances after each test case."""
+        db.session.remove()
+        db.drop_all()
+        self.app_context.pop()
+
+    def api_email_auth_headers(self, email, password):
+        """Setup an authorization header with email and password."""
+        return {
+            'Authorization': 'Basic ' + b64encode(
+                (email + ':' + password).encode('utf-8')).decode('utf-8'),
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+
+    def api_token_auth_headers(self, token):
+        """Setup an authorization header with a token."""
+        return {
+            'Authorization': 'Basic ' + b64encode(
+                (token).encode('utf-8')).decode('utf-8'),
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+
+    def test_verify_password_with_token(self):
+        """Test verify_password function with valid token."""
+        self.user.save()
```

Comment:
Since I didn't need the user in the database for all the test case, I felt it would be better to save the user in test cases where they are needed but I can see I have only one test case in which the user is not needed in the database so I can just as well ignore it that one time.